waitfor.go

```diff
@@ -187,6 +192,104 @@ func (client Client) WaitForLKEClusterStatus(ctx context.Context, clusterID int,
 	}
 }
 
+// LKEClusterPollOptions configures polls against LKE Clusters.
+type LKEClusterPollOptions struct {
+	// TimeoutSeconds is the number of Seconds to wait for the poll to succeed
+	// before exiting.
+	TimeoutSeconds int
+
+	// TansportWrapper allows adding a transport middleware function that will
+	// wrap the LKE Cluster client's undelying http.RoundTripper.
+	TransportWrapper func(http.RoundTripper) http.RoundTripper
+}
+
+func getLKEClusterClientset(
+	ctx context.Context,
+	client *Client,
+	clusterID int,
+	transportWrapper func(http.RoundTripper) http.RoundTripper,
+) (kubernetes.Clientset, error) {
+	resp, err := client.GetLKEClusterKubeconfig(ctx, clusterID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Kubeconfig for LKE cluster %d: %s", clusterID, err)
+	}
+
+	kubeConfigBytes, err := base64.StdEncoding.DecodeString(resp.KubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode kubeconfig: %s", err)
+	}
+
+	clientset, err := kubernetes.BuildClientsetFromConfig(kubeConfigBytes, transportWrapper)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client for LKE cluster %d: %s", clusterID, err)
+	}
+	return clientset, nil
+}
+
+// WaitForLKEClusterConditions waits for the given LKE conditions to be true
+func (client Client) WaitForLKEClusterConditions(
+	ctx context.Context,
+	clusterID int,
+	options LKEClusterPollOptions,
+	conditions ...condition.ClusterConditionFunc,
+) error {
+	ctx, cancel := context.WithCancel(ctx)
+	if options.TimeoutSeconds != 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(options.TimeoutSeconds)*time.Second)
+	}
+	defer cancel()
+
+	ticker := time.NewTicker(client.millisecondsPerPoll * time.Millisecond)
+	defer ticker.Stop()
+
+	var prevLog string
+	var clientset kubernetes.Clientset
+
+	clientReady := func(ctx context.Context, c kubernetes.Clientset) (bool, error) {
+		var err error
+		if clientset == nil {
+			clientset, err = getLKEClusterClientset(ctx, &client, clusterID, options.TransportWrapper)
```

Comment:
In my opinion each of these condition functions should use the `c` which is passed to them, not construct any new clientsets. 

@Charliekenney23 do you think it's strange that the first condition constructs the clientset for all of the other conditions, by abusing a closure?